frontend/packages/dev-console/src/components/pipelines/utils/triggers.ts

```diff
@@ -149,23 +149,27 @@ export const usePipelineTriggerTemplateNames = (
   }, []);
 };
 
-export const useEventListenerTriggerTemplateNames = (
+export const useEventListenerURL = (
   eventListener: EventListenerKind,
-): RouteTemplate[] | null => {
-  const {
-    metadata: { namespace },
-  } = eventListener;
-
+  namespace: string,
+): string | null => {
   const [route, routeLoaded] = useK8sGet<RouteKind>(
     RouteModel,
     getEventListenerGeneratedName(eventListener),
     namespace,
   );
+
+  return routeLoaded ? getRouteWebURL(route) : null;
+};
+
+export const useEventListenerTriggerTemplateNames = (
+  eventListener: EventListenerKind,
+): RouteTemplate[] | null => {
   return eventListener.spec.triggers.reduce(
```

Comment:
Two things:

1. I think this should have always been a `.map` (missed it during my original review I suspect)
2. You should just change this logic to render resources - see how other resources are rendered

If you look at how the TriggerBindings are rendered on this page, you can see they are just a list of resources (from `DynamicResourceLinkList`). Also if you look at where `ResourceLinkList` (which calls `DynamicResourceLinkList`) is used, you can see it's also used for Tasks in a similar fashion.

You reduced the logic down to what you needed, but you didn't shake the previous design. This hook is designed for that page, you should morph it into what you need and make use of other components that render the list out as you need (such as `ResourceLinkList`).

Reply:
Updated the hook to return `ResourceModelLink[]` and reused `ResourceLinkList`.